src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -179,49 +260,121 @@ private static CollectionSettings createCollectionSettings(
       SimilarityFunction function,
       String comment,
       ObjectMapper objectMapper) {
-
     if (comment == null || comment.isBlank()) {
-      return new CollectionSettings(
-          collectionName, vectorEnabled, vectorSize, function, null, null, null);
+      // vector column exists -> vectorEnabled, vectorSize
+      // vector index exists -> similarityFunction
+      if (vectorEnabled) {
```

Comment:
What is `vector index exists -> similarityFunction`?

Reply:
sorry for the confusion
Basically, we extract information such as similarityFunction, vectorEnabled, vectorSize from vector column and vector index.